src/torch/SIRF_torch.py

```diff
@@ -0,0 +1,123 @@
+try:
+    import torch
+except ImportError:
+    raise ImportError('Failed to import torch. Please install PyTorch first.')
+
+# based on 
+# https://github.com/educating-dip/pet_deep_image_prior/blob/main/src/deep_image_prior/torch_wrapper.py
+
+
+class _objectiveFunctionModule3D(torch.autograd.Function):
+    @staticmethod
+    def forward( ctx, x, image_template, sirf_obj):
+        ctx.device = x.device
+        ctx.sirf_obj = sirf_obj
+        ctx.image_template = image_template
+        ctx.x = x.detach().cpu().numpy().squeeze()
+        ctx.x = ctx.image_template.fill(ctx.x)
```

Comment:
these operations are going to occur many places, and are prone to changes, so I suggest to have a function
```python
def fill_from_torch(sirf_object, tensor)
```
I suppose we could even put this somewhere in the main SIRF `DataContainer` as a member as well, but let's leave that for later.